core/vm/context.go

```diff
@@ -185,3 +188,57 @@ func VerifySealFn(ref *types.Header, chain ChainContext) func(*types.Header) boo
 		return chain.Engine().VerifySeal(nil, header) == nil
 	}
 }
+
+func getTobinTax(evm *EVM, sender common.Address) (numerator *big.Int, denominator *big.Int, reserveAddress *common.Address, err error) {
+	reserveAddress, err = GetRegisteredAddressWithEvm(params.ReserveRegistryId, evm)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	ret, _, err := evm.Call(AccountRef(sender), *reserveAddress, params.TobinTaxFunctionSelector, params.MaxGasForGetOrComputeTobinTax, big.NewInt(0))
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// Expected size of ret is 64 bytes because getOrComputeTobinTax() returns two uint256 values,
+	// each of which is equivalent to 32 bytes
+	if binary.Size(ret) != 64 {
+		return nil, nil, nil, goerrors.New("Length of tobin tax not equal to 64 bytes")
+	}
+	numerator = new(big.Int).SetBytes(ret[0:32])
+	denominator = new(big.Int).SetBytes(ret[32:64])
+	if denominator.Cmp(common.Big0) == 0 {
+		return nil, nil, nil, goerrors.New("Tobin tax denominator equal to zero")
+	}
+	if numerator.Cmp(denominator) == 1 {
+		return nil, nil, nil, goerrors.New("Tobin tax numerator greater than denominator")
+	}
+	return numerator, denominator, reserveAddress, nil
+}
+
+// TobinTransfer performs a transfer that may take a tax from the sent amount and give it to the reserve.
+// If the calculation or transfer of the tax amount fails for any reason, the regular transfer goes ahead.
+// NB: Gas is not charged or accounted for this calculation.
+func TobinTransfer(evm *EVM, sender, recipient common.Address, amount *big.Int) {
```

Comment:
Hey @mcortesi, this PR doesn't seem to be actually removing knowledge of tobin tax at the package level, I think it would be nice if we could really decouple the vm from this special logic.

I created a PR showing how this could work here - #1463